ipn/ipnstate/ipnstate.go

```diff
@@ -53,6 +53,9 @@ type Status struct {
 	// has MagicDNS enabled.
 	MagicDNSSuffix string
 
+	// MagicDNSEnabled is whether or not the network has MagicDNS enabled.
```

Comment:
You should probably clarify that even if this is true, MagicDNS is not necessarily enabled on this node if `--accept-dns=false`. That is, this is purely about whether it's enabled on the tailnet.

Reply:
Good point!

Reply:
in fact maybe the field should be more clearly named too then.

Or maybe we should make a Tailnet-specific JSON object to put all such fields.

Reply:
(because who really reads docs? people just run `tailscale status --json` and guess what things mean usually)

Reply:
Any objection to this kind of structure within the `Status` struct?

```go
Tailnet struct {
  Name string
  MagicDNSEnabled bool
}
```

Reply:
Don't do an anonymous struct in this case. Make a named type.

Reply:
Yeah, I'll make a named struct. Just wanted your thoughts on the structure.

Reply:
> (because who really reads docs? people just run `tailscale status --json` and guess what things mean usually)
 
Hey there ^^ ! I've just landed here searching for a bug in our integration tests on tailscale < 1.22